de.metas.contracts/src/main/java/de/metas/flatrate/invoicecandidate/spi/impl/FlatrateTermHandler.java

```diff
@@ -309,7 +309,7 @@ public void setOrderedData(final I_C_Invoice_Candidate ic)
 
 		ic.setDateOrdered(getDateOrdered(term));
 
-		if (X_C_Flatrate_Term.CONTRACTSTATUS_Gekuendigt.equals(term.getContractStatus()))
+		if (X_C_Flatrate_Term.CONTRACTSTATUS_Gekuendigt.equals(term.getContractStatus()) && term.isCloseInvoiceCandidate()) // terminate invoice candidates only when the flag is set
```

Comment:
i think in this case it would be much better to call `IInvoiceCandBL.closeInvoiceCandidate()`